hyperspy/signal.py

```diff
@@ -3496,37 +3496,32 @@ def _make_sure_data_is_contiguous(self):
                          "for more information.".format(self))
             self.data = np.ascontiguousarray(self.data)
 
-    def _iterate_signal(self):
-        """Iterates over the signal data.
+    def _iterate_signal(self, iterpath=None):
+        """Iterates over the signal data. It is faster than using the signal
+        iterator.
```

Comment:
What is "the signal iterator" that this is faster than? Assuming you mean `for si in s`?
How about `"""Iterates over the signal data. It is faster than iterating over a signal, since it returns the array directly and not a HyperSpy signal.`

Reply:
Actually, it would be useful to explain why it is faster! I don't remember right now and I will need to look it up again.

Reply:
Yes, that was my best guess :p

Reply:
> What is "the signal iterator" that this is faster than?

The `BaseSignal` class has an iterator implementation, i. e. `__iter__` and `__next__` are implemented, see https://docs.python.org/3/library/stdtypes.html#iterator-types for more explanation.